File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.py
import math


def solution(r1, r2):
    answer = 0
    
    int_sqrt_r1 = math.ceil(r1 * (1/math.sqrt(2)))
    int_sqrt_r2 = math.floor(r2 * (1/math.sqrt(2)))
    
    for y in range(0, int_sqrt_r2 + 1):
        if y > r1:
            min_x = y
        else:
            min_x = max(math.ceil(math.sqrt(r1 ** 2 - y ** 2)), y)
        max_x = math.floor(math.sqrt(r2 ** 2 - y ** 2))
        x_count = max_x - min_x + 1
        if y == 0:
            answer += x_count
        else:
            answer += 2 * x_count
    
    answer -= int_sqrt_r2 - int_sqrt_r1 + 1
    
    return answer * 4

File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.java
class Solution {
    public long solution(int r1, int r2) {
        long answer = 0L;
        
        for (int i = 0; i < r1; i++){
            answer += (long) Math.floor(Math.sqrt((long)r2 * r2 - (long) i * i)) - Math.ceil(Math.sqrt((long)r1 * r1 - (long) i * i)) + 1L;
        }
        for (int i = r1; i < r2; i++){
            answer += (long) Math.floor(Math.sqrt((long) r2 * r2 - (long) i * i));
        }
        
        return answer * 4L;
    }
}